packages/flutter/lib/src/material/floating_action_button.dart

```diff
@@ -319,10 +314,40 @@ class FloatingActionButton extends StatelessWidget {
 
   final BoxConstraints _sizeConstraints;
 
+  static const double _defaultElevation = 6;
+  static const double _defaultHighlightElevation = 12;
+  static const ShapeBorder _defaultShape = CircleBorder();
+  static const ShapeBorder _defaultExtendedShape = StadiumBorder();
+
   @override
   Widget build(BuildContext context) {
     final ThemeData theme = Theme.of(context);
-    final Color foregroundColor = this.foregroundColor ?? theme.accentIconTheme.color;
+    final FloatingActionButtonThemeData floatingActionButtonTheme = theme.floatingActionButtonTheme;
```

Comment:
I think you should use the `FloatingActionButtonThemeData.of(context)` constructor here.

Reply:
Why?

Reply:
I think it was for consistency. Although, I do not think the .of(context) methods are even necessary for non-inherited widgets. But the other non-inherited widget FooTheme classes are already doing it.